spv/verifypayment.go

```diff
@@ -89,32 +93,50 @@ func (v *verifier) verifyTxAnchor(ctx context.Context, payment *Envelope) (bool,
 	return valid, nil
 }
 
-func (v *verifier) verifyUnconfirmedTx(tx *bt.Tx, payment *Envelope) (bool, error) {
+func (v *verifier) verifyUnconfirmedTx(ctx context.Context, tx *bt.Tx, payment *Envelope) (bool, error) {
 	// If no tx inputs have been provided, fail and error
 	if len(tx.Inputs) == 0 {
-		return false, ErrNoTxInputsToVerify
+		return false, errors.Wrapf(ErrNoTxInputsToVerify, "tx %s has no inputs to verify", tx.TxID())
 	}
 
-	for _, input := range tx.Inputs {
-		parent, ok := payment.Parents[input.PreviousTxIDStr()]
-		if !ok {
-			return false, ErrNotAllInputsSupplied
-		}
-
-		parentTx, err := bt.NewTxFromString(parent.RawTx)
-		if err != nil {
-			return false, err
-		}
-
-		// If the input is indexing an output that is out of bounds, fail and error
-		if int(input.PreviousTxOutIndex) > len(parentTx.Outputs)-1 {
-			return false, ErrInputRefsOutOfBoundsOutput
-		}
-
-		output := parentTx.Outputs[int(input.PreviousTxOutIndex)]
+	// perform the script validations in parallel
+	errs, _ := errgroup.WithContext(ctx)
+	for i := range tx.Inputs {
+		idx := i // copy current value of i for concurrent use
+		errs.Go(func() error {
+			input := tx.InputIdx(idx)
+
+			parent, ok := payment.Parents[input.PreviousTxIDStr()]
+			if !ok {
+				return errors.Wrapf(ErrNotAllInputsSupplied, "tx %s is missing input %d in its parents' envelope", tx.TxID(), idx)
+			}
+
+			parentTx, err := bt.NewTxFromString(parent.RawTx)
+			if err != nil {
+				return err
+			}
+
+			output := parentTx.OutputIdx(int(input.PreviousTxOutIndex))
+			// If the input is indexing an output that is out of bounds, fail and error
+			if output == nil {
+				return errors.Wrapf(ErrInputRefsOutOfBoundsOutput, "tx %s input %d is referencing an out of bounds output", tx.TxID(), idx)
+			}
+
+			if err = v.eng.Execute(interpreter.ExecutionParams{
+				PreviousTxOut: output,
+				InputIdx:      idx,
+				Tx:            tx,
+				Flags:         interpreter.ScriptEnableSighashForkID | interpreter.ScriptUTXOAfterGenesis,
```

Comment:
@jadwahab would you know when the after genesis flag should be enabled? Also, what other flags would you like to see running here at all times?